### ISSUE TITLE
Addition of parametric and non-parametric `TransferFunction`

### ISSUE THREAD

**Issue by artpelling** [pymor/pymor#1918]
**Describe the bug**
I ran into a bug where I added a parametric and non-parametric `TransferFunction` and `FactorizedTransferFunction` and the result had an empty `parameters` attribute that lead to errors being thrown.

**To Reproduce**
A minimal (non-working) example:

```python
from pymor.models.transfer_function import TransferFunction
from pymor.parameters.base import Parameters

tf1 = lambda s, mu: s*mu['p']
tf2 = lambda s: s

sys1 = TransferFunction(1, 1, tf1, parameters={'p': 1})
sys2 = TransferFunction(1, 1, tf2)

# this works
(sys1 + sys2).eval_tf(1.0, mu=3.0)

# this doesn't
(sys2 + sys1).eval_tf(1.0, mu=3.0)
```

**Expected behavior**
I would expect both lines to work the same.

**Comment by artpelling** [comment]
I'm not sure how to fix this properly, otherwise I would have opened a PR. Maybe you have an idea @pmli?

**Comment by pmli** [comment]
Good catch. Maybe it's enough to set the parameters of the addition as the union of parameters.

Replacing
https://github.com/pymor/pymor/blob/f1677cf9959d59e10b0fd95d3a911c9979e4bd5f/src/pymor/models/transfer_function.py#L441
by
```python
return self.with_(tf=tf, dtf=dtf, parameters=self.parameters | other.parameters)
```
might work.

**Comment by sdrave** [comment]
One could also consider if it makes sense to use pyMOR `Functions`, at least internally. This would automatically take care of the parameterization.

**Comment by pmli** [comment]
> One could also consider if it makes sense to use pyMOR `Functions`, at least internally. This would automatically take care of the parameterization.

Yes, indeed, that might be a better approach.
